packages/block-editor/src/components/use-settings/index.js

```diff
@@ -0,0 +1,272 @@
+	'spacing.customMargin': 'spacing.margin',
+	'spacing.customPadding': 'spacing.padding',
+	'typography.customLineHeight': 'typography.lineHeight',
+};
+
+/**
+ * Remove `custom` prefixes for flags that did not land in 5.8.
+ *
+ * This provides continued support for `custom` prefixed properties. It will
+ * be removed once third party devs have had sufficient time to update themes,
+ * plugins, etc.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/34485
+ *
+ * @param {string} path Path to desired value in settings.
+ * @return {string}     The value for defined setting.
+ */
+const removeCustomPrefixes = ( path ) => {
+	return prefixedFlags[ path ] || path;
+};
+
+/**
+ * For settings like `color.palette`, which have a value that is an object
+ * with `default`, `theme`, `custom`, with field values that are arrays of
+ * items, merge these three arrays into one and return it. The calculation
+ * is memoized so that identical input values produce identical output.
+ * @param {Object} value Object to merge
+ * @return {Array} Array of merged items
+ */
+function mergeOrigins( value ) {
+	let result = mergeCache.get( value );
+	if ( ! result ) {
+		result = [ 'default', 'theme', 'custom' ].flatMap(
+			( key ) => value[ key ] ?? []
+		);
+		mergeCache.set( value, result );
+	}
+	return result;
+}
+const mergeCache = new WeakMap();
+
+/**
+ * Hook that retrieves the given settings for the block instance in use.
+ *
+ * It looks up the settings first in the block instance hierarchy.
+ * If none are found, it'll look them up in the block editor settings.
+ *
+ * @param {string[]} paths The paths to the settings.
+ * @return {any[]} Returns the values defined for the settings.
+ * @example
+ * ```js
+ * const [ fixed, sticky ] = useSettings( 'position.fixed', 'position.sticky' );
+ * ```
+ */
+export function useSettings( ...paths ) {
+	const { name: blockName, clientId = null } = useBlockEditContext();
+
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	paths = useMemo( () => paths, paths );
+
+	return useSelect(
+		( select ) => {
+			const candidates = clientId
+				? [
+						clientId,
+						...select( blockEditorStore ).getBlockParents(
+							clientId,
+							/* ascending */ true
+						),
+				  ].filter( ( candidateClientId ) => {
+						const candidateBlockName =
+							select( blockEditorStore ).getBlockName(
+								candidateClientId
+							);
+						return hasBlockSupport(
+							candidateBlockName,
+							'__experimentalSettings',
+							false
+						);
+				  } )
+				: [];
+
+			return paths.map( ( path ) => {
+				if ( blockedPaths.includes( path ) ) {
+					// eslint-disable-next-line no-console
+					console.warn(
+						'Top level useSetting paths are disabled. Please use a subpath to query the information needed.'
+					);
+					return undefined;
+				}
+
+				// 0. Allow third parties to filter the block's settings at runtime.
+				let result = applyFilters(
+					'blockEditor.useSetting.before',
```

Comment:
There's no need to introduce a bulk settings filter right?

Reply:
Certainly not, it would be counterproductive. If you want to modify a setting with a plugin, you want to register only one filter, not two of them, both `useSetting` and `useSettings`.